File: src/src_platform_ava/fmg/ava/utils/ProjSettings.cs
using Avalonia.Threading;
using Fmg.Common;
using Fmg.Common.UI;
using Fmg.Ava.Img;

namespace Fmg.Ava.Utils {

    class ProjSettings : AProjSettings {

        static ProjSettings() {
            UiInvoker.Deferred = doRun => Dispatcher.UIThread.InvokeAsync(() => doRun(), DispatcherPriority.Normal);
            UiInvoker.Animator = () => Animator.Singleton;
            UiInvoker.TimerCreator = () => new Timer();
        }

        public static void Init() {
            // implicit call static block
        }

    }

}

File: src/src_platform_jfx/fmg/jfx/utils/ProjSettings.java
package fmg.jfx.utils;

import fmg.common.AProjSettings;
import fmg.common.ui.UiInvoker;
import fmg.jfx.img.Animator;

public final class ProjSettings extends AProjSettings {

    private ProjSettings() {}

    static {
        UiInvoker.DEFERRED = javafx.application.Platform::runLater;
        UiInvoker.ANIMATOR = Animator::getSingleton;
        UiInvoker.TIMER_CREATOR = Timer::new;

//        try {
//            javafx.scene.layout.Region region = new javafx.scene.layout.Pane();
//            javafx.scene.layout.Background bk = region.getBackground(); // its NULL :(
//            java.util.List<javafx.scene.layout.BackgroundFill> fills = bk.getFills();
//            javafx.scene.layout.BackgroundFill fill = fills.iterator().next();
//            javafx.scene.paint.Paint p = fill.getFill();
//            Logger.info("ButtonColor = " + p);
//        } catch (Exception e) {
//            fmg.core.mosaic.MosaicDrawModel.DefaultBkColor = fmg.common.Color.Gray().brighter();
//        }
    }

    public static void init() {
        // implicit call static block
    }

}
